File: app/src/main/java/net/mmho/photomap2/ThumbnailImageView.java
package net.mmho.photomap2;

import android.content.Context;
import android.util.AttributeSet;

public class ThumbnailImageView extends LoadableImageView{

    public ThumbnailImageView(Context context) {
        this(context,null);
    }

    public ThumbnailImageView(Context context, AttributeSet attrs) {
        this(context, attrs,0);
    }

    public ThumbnailImageView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        super.thumbnail = true;
    }


    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        setMeasuredDimension(widthMeasureSpec,widthMeasureSpec);
    }

}

File: app/src/main/java/net/mmho/photomap2/ThumbnailImageView.kt
package net.mmho.photomap2

import android.content.Context
import android.util.AttributeSet

class ThumbnailImageView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0)
        : LoadableImageView(context, attrs, defStyle) {

    init {
        super.thumbnail = true
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        setMeasuredDimension(widthMeasureSpec, widthMeasureSpec)
    }

}
